### 后端返回异常对象的情况处理
```js
function judgeErrorByResponseType(response: any) {
    const { data } = response;
    return new Promise((resolve, reject) => {
      if (data) {
        // 此处拿到的data才是blob
        const reader = new FileReader();
        reader.onload = () => {
          const { result } = reader;
          const errorInfos = JSON.parse(result as string);
          resolve(errorInfos);
        };
        reader.onerror = err => {
          resolve(err);
        };
        reader.readAsText(data);
      }
    });
  }

try {
    const { response, data } = await ...;
    if (response && data) {
        const match = response.headers.get('content-disposition')?.match(/filename=(.*);?/);
        if (match) {
        downloadBlob(match ? match[1] : '未知文件', data);
        } else {
        const err: any = await judgeErrorByResponseType({ data });
        err && message.error(err?.message);
        }
    }
    } catch (error: any) {
    const err: any = await judgeErrorByResponseType(error);
    err && message.error(err?.message);
    throw new Error(error);
    }
```